wallet-api/src/Ledger/Crypto.hs

```diff
@@ -1,35 +1,91 @@
+{-# LANGUAGE DataKinds          #-}
 {-# LANGUAGE DeriveAnyClass     #-}
 {-# LANGUAGE DeriveGeneric      #-}
 {-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE OverloadedStrings  #-}
 {-# LANGUAGE TemplateHaskell    #-}
 module Ledger.Crypto where
 
-import           Codec.Serialise.Class        (Serialise)
-import           Control.Newtype.Generics     (Newtype)
-import           Data.Aeson                   (FromJSON, ToJSON)
-import           Data.Swagger.Internal.Schema (ToSchema)
-import           GHC.Generics                 (Generic)
-import           Language.PlutusTx.Lift       (makeLift)
+import           Codec.Serialise.Class      (Serialise)
+import           Control.Newtype.Generics   (Newtype)
+import qualified Crypto.ECC.Ed25519Donna    as ED25519
+import           Crypto.Error               (throwCryptoError)
+import           Data.Aeson                 (FromJSON (parseJSON), FromJSONKey, ToJSON (toJSON), ToJSONKey)
+import qualified Data.Aeson                 as JSON
+import qualified Data.Aeson.Extras          as JSON
+import qualified Data.ByteArray             as BA
+import qualified Data.ByteString            as BS
+import qualified Data.ByteString.Lazy       as BSL
+import           Data.Swagger               (ToSchema (declareNamedSchema), byteSchema)
+import           Data.Swagger.Internal
+import           GHC.Generics               (Generic)
+import           KeyBytes                   (KeyBytes)
+import qualified KeyBytes                   as KB
+import qualified Language.PlutusTx.Builtins as Builtins
+import           Language.PlutusTx.Lift     (makeLift)
+import           Ledger.TxId
+import           Servant.API                (FromHttpApiData (parseUrlPiece), ToHttpApiData (toUrlPiece))
 
 -- | A cryptographic public key.
-newtype PubKey = PubKey { getPubKey :: Int }
+newtype PubKey = PubKey { getPubKey :: KeyBytes }
     deriving (Eq, Ord, Show)
     deriving stock (Generic)
-    deriving anyclass (ToSchema, ToJSON, FromJSON, Newtype)
+    deriving anyclass (ToSchema, ToJSON, FromJSON, Newtype, ToJSONKey, FromJSONKey)
     deriving newtype (Serialise)
-
 makeLift ''PubKey
 
+-- | A cryptographic private key.
+newtype PrivateKey = PrivateKey { getPrivateKey :: KeyBytes }
```

Comment:
I changed it to use the `PublicKey` and `PrivateKey` types everywhere so `KeyBytes` type isn't actually necessary anymore. I think that's better than having a 64-byte `ByteString` with two dedicated segments.

Reply:
Okay, but the `PublicKey` should definitely not contain the private key in case we serialize it.

Reply:
Hm, I just realised something. We do actually serialise private keys now. (To identify wallets in the Playground). But I think I'll change it so that wallets are still `Int`s and to get a wallet's private key we look up the index in the list of known private keys.

Reply:
I think that's benign, since it's just in the playground which isn't meant to be secure.

Reply:
Yeah but keeping wallets as `Int`s has the advantage that I don't need to update the playground code ;)